src/components/Button/Button.tsx

```diff
@@ -54,6 +56,8 @@ export interface Props {
   ariaControls?: string;
   /** Tells screen reader the controlled element is expanded */
   ariaExpanded?: boolean;
+  /** Tells screen reader the element is pressed */
+  ariaPressed?: boolean;
```

Comment:
The reason I did in this PR is because it's resurfacing because of this: https://github.com/Shopify/web/issues/9320. Did feel like it was a big enough change to warrant its own PR. Do you feel it is?

Reply:
Probably should be its own PR; pretty sure the pressed state needs its own visual change. We actually have something in Polaris Rails for this: https://github.com/Shopify/polaris-rails/pull/993

Reply:
For the current web requirement, visually there is no difference. Are you ok to leave this here but tackle the visual in another PR? Just to manage scope while I'm working on the product pages.